### PR TITLE
feat: implement testing of `Filecoin.ChainGetBlockMessages` API

### PR DESCRIPTION
**Pull request by elmattic**
## Summary of changes

Changes introduced in this pull request:

- Add `Filecoin.ChainGetBlockMessages` testing in `forest-tool api compare`

## Reference issue to close (if applicable)

Part of https://github.com/ChainSafe/forest/issues/3639

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,10 @@
 
 ### Added
 
+- [#3675](https://github.com/ChainSafe/forest/pull/3675) Implement support of
+  testing `Filecoin.ChainGetBlockMessages` endpoint in `forest-tool api compare`
+  subcommand. Add `forest-cli chain block-messages` subcommand.
```
Comment (lemmih) [inline CHANGELOG.md:34]: Testing is not user-visible. The target audience for the CHANGELOG are end-users, not other Forest developers.

File: src/cli/subcommands/chain_cmd.rs [hunk 2]
```diff
@@ -12,6 +12,12 @@ use super::{print_pretty_json, print_rpc_res_cids};
 
 #[derive(Debug, Subcommand)]
 pub enum ChainCommands {
+    /// Retrieves and prints messages stored in the block specified by the given CID.
+    BlockMessages {
+        #[arg(short)]
+        cid: Cid,
+    },
+
```
Comment (lemmih) [inline src/cli/subcommands/chain_cmd.rs:20]: Why did you implement this command? What is its purpose? If it is useful, why doesn't Lotus have a similar command?

**Review by lemmih — CHANGES_REQUESTED** [review]

File: src/cli/subcommands/chain_cmd.rs (hunk 2 above)
Comment (elmattic) [reply src/cli/subcommands/chain_cmd.rs:20]: Right, Lotus doesn't have this command. It only uses the `Filecoin.ChainGetBlockMessages` in its `chain get-block subcommand`. I can remove it.

**Review by lemmih — APPROVED** [review]
Just to confirm: `Filecoin.ChainGetBlockMessages` now shows up as `Valid` for both Forest and Lotus?

**Comment by lemmih** [comment]
Oh, btw, the PR description needs to be updated.

**Comment by lemmih** [comment]
And this PR is part of #3639

**Comment by elmattic** [comment]
> Just to confirm: `Filecoin.ChainGetBlockMessages` now shows up as `Valid` for both Forest and Lotus?

Yes, I'm seeing that:
```
| RPC Method                            | Forest              | Lotus         |
|---------------------------------------|---------------------|---------------|
| Filecoin.ChainGetBlock                | Valid               | Valid         |
| Filecoin.ChainGetBlockMessages        | Valid               | Valid         |
| Filecoin.ChainGetGenesis              | Valid               | Valid         |
...
```

**Comment by lemmih** [comment]
> > Just to confirm: `Filecoin.ChainGetBlockMessages` now shows up as `Valid` for both Forest and Lotus?
> 
> Yes, I'm seeing that:
> 
> ```
> | RPC Method                            | Forest              | Lotus         |
> |---------------------------------------|---------------------|---------------|
> | Filecoin.ChainGetBlock                | Valid               | Valid         |
> | Filecoin.ChainGetBlockMessages        | Valid               | Valid         |
> | Filecoin.ChainGetGenesis              | Valid               | Valid         |
> ...
> ```

Excellent!

**Review by sudo-shashank — APPROVED** [review]

**Pull request merged**